native/src/main/java/io/ballerina/stdlib/task/TokenAcquisition.java

```diff
@@ -0,0 +1,189 @@
+                                                     BString id, boolean tokenAcquired, int livenessInterval) {
+        DatabaseConfig dbConfig = new DatabaseConfig(
+                databaseConfig.getStringValue(DB_HOST).getValue(), databaseConfig.getStringValue(DB_USER).getValue(),
+                databaseConfig.getStringValue(DB_PASSWORD).getValue(), databaseConfig.getIntValue(DB_PORT).intValue(),
+                databaseConfig.getStringValue(DATABASE).getValue(), null
+        );
+        try {
+            String instanceId = id.getValue();
+            String jdbcUrl = String.format(JDBC_URL, dbConfig.host(), dbConfig.port(), dbConfig.database());
+            Connection connection = DriverManager.getConnection(jdbcUrl, dbConfig.user(), dbConfig.password());
+            connection.setAutoCommit(false);
+            tokenAcquired = attemptTokenAcquisition(connection, instanceId, tokenAcquired, livenessInterval);
+            connection.commit();
+            startHealthCheckUpdater(connection, instanceId, 1);
+            if (!tokenAcquired) {
+                upsertToken(connection, instanceId);
+            }
+            return generateResponse(tokenAcquired, livenessInterval, instanceId, connection);
+        } catch (Exception e) {
+            throw Utils.createTaskError(e.getMessage());
+        }
+    }
+
+    private static BMap<BString, Object> generateResponse(boolean tokenAcquired, int interval, String myInstanceId,
+                                                          Connection connection) {
+        BMap<BString, Object> response = ValueCreator.createMapValue();
+        response.put(STATUS, tokenAcquired ? ACTIVE_STATUS : STANDBY_STATUS);
+        response.put(INSTANCE_ID, StringUtils.fromString(myInstanceId));
+        response.put(TOKEN_HOLDER, tokenAcquired);
+        response.put(CONNECTION, connection);
+        response.put(LIVENESS_INTERVAL, interval);
+        return response;
+    }
+
+    public static void upsertToken(Connection connection, String myInstanceId) throws SQLException {
+        String sql = "INSERT INTO token_holder(token_id, is_active) VALUES (?, false) " +
+                "ON DUPLICATE KEY UPDATE is_active = VALUES(is_active)";
+
+        try (PreparedStatement stmt = connection.prepareStatement(sql)) {
+            stmt.setString(1, myInstanceId);
+            stmt.executeUpdate();
+            connection.commit();
+        }
+    }
+
+
+    public static boolean attemptTokenAcquisition(Connection connection, String myInstanceId,
+                                                  boolean acquired, int livenessInterval) throws SQLException {
+        boolean tokenAcquired = acquired;
+        String checkSql = "SELECT token_id FROM token_holder WHERE is_active = true";
+
+        try (PreparedStatement stmt = connection.prepareStatement(checkSql)) {
+            ResultSet rs = stmt.executeQuery();
+
+            if (!rs.next()) {
```

Comment:
Here, checking the active token holder and updating the token isn't atomic. Right? This may result in a race condition, allowing two instances to acquire the token simultaneously.